### DISCUSSION TITLE
Tinyusb on windows10

### DISCUSSION THREAD

**Discussion by fanofbluetooth** [adafruit/Adafruit_nRF52_Bootloader#263] [Q&A]
Adafruit nRF52 bootloader's tinyusb cdc isn't detected on Windows or shows device response failed sometimes on some devices.

I developed a custom pcb using a modified Adafruit nRF52 bootloader based on Adafruit itsybitsy nRF52840 express.

The bootloader and firmware works well most of the time, but sometimes it suddenly stops detecting that the usb is plugged in or detects unknown device (device response failed).
- Tried reconnecting, reflashing the bootloader with j-link, and rebooting the pc without a successful result.
- Plugging into another usb port or switching to another pc, it is then able to get response again.

Any thoughts about resolving this issue?

**Comment by hathach** [comment]
it is basically custom firmware on custom hardware, and have no meaningful to analyze the issue. I couldn't anyway, since it is custom hardware. My advice is 
- test known-good firmware on custom hardware
- test custom firmware on known-good hardware e.g Adafruit boards
